### DISCUSSION TITLE
RUN_INTERVAL in msg websocket

### DISCUSSION THREAD

**Discussion by Menzer05** [actix/actix-web#2260] [General]
Please, is there a way to exit a RUN_INTERVAL without stopping the context ??

my example:
else if &msg[..4] == "graf"{
                    let intv = &msg[4..];
                    let tps = intv.parse::<u64>().unwrap();
     
                    let mut i: i8 = 0;
                        ctx.run_interval(Duration::from_secs(tps), move |act, ctx| {
                        ctx.text(bindex);
                        i += 1;
                        if i == 5 {
                          ctx.stop();
                         // ctx.terminate();
                        
                        }
                        
                         });                                                         
                }

**Comment by thalesfragoso** [comment]
`run_interval` returns a `SpawnHandle`, you should be able to store it somewhere (inside the actor) and then call the `cancel_future` method of `ctx`.

**Reply by Menzer05** [reply]
thanks for your help Thales, but it didn't work for me so if i understood correctly there is no other way to pause the context ???
